app/brave_generated_resources.grd

```diff
@@ -876,6 +876,21 @@ Or change later at <ph name="SETTINGS_EXTENIONS_LINK">$2<ex>brave://settings/ext
       <message name="IDS_BRAVE_IPFS_LEARN_MORE">
         Learn more about IPFS and privacy
       </message>
+      <message name="IDS_BRAVE_IPFS_INFOBAR_TEXT" desc="Text that promotes turning on autoredirect to the configured gateway setting">
+        Would you like to handle content-addressed resources with native IPFS support (ipfs:// and ipns://) built into Brave?
```

Comment:
@rmcfadden3 any thoughts on this?

Maybe just move the parents like so:

`Would you like to handle content-addressed resources (ipfs:// and ipns://) with native IPFS support built into Brave?`

or

`Would you like Brave to natively handle content-addressed resources such as ipfs:// and ipns://?`

or

`Would you like to use native IPFS support built into Brave to handle content-addressed resources (ipfs:// and ipns://)?`

Reply:
Per Slack convo with @mkarolin …

Let's avoid phrasing like "content-addressed resources" as they may not be fully understood by all users. Instead, we landed on the following:

`If a site can be opened with an IPFS link (like ipfs:// or ipns://), would you like to open it with Brave's built-in IPFS support?`

Reply:
I am afraid these changed sentences bury the lead – this feature is not about enabling `ipfs://` but UPGRADING `https://` to `ipfs://` and `ipns://` when possible.



"IPFS resources" on the web could be `https://` URLs of public gateways like:

- https://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.w3s.link
- https://cloudflare-ipfs.com/ipfs/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi

Or [DNSLink](https://dnslink.dev/) website like:

- https://en.wikipedia-on-ipfs.org

Ref. https://docs.ipfs.tech/how-to/address-ipfs-on-web

Brave detects these resources and instead of loading them over HTTP, it uses local IPFS node and presents them as loaded via `ipfs://`  or `ipns://` in the address bar, to indicate native IPFS was used and the hashes were verified.


With that in mind, perhaps below will work better?

```suggestion
        Would you like to upgrade IPFS resources like this and load them using the native support (ipfs:// and ipns://) built into Brave?
```

Reply:
Last version is good, would just like one change:

"opened with an IPFS link" is not ideal, given how IPFS works. just making a link is not enough.

"If a site is available on IPFS" or "on the IPFS network"  or something like that would be better.

Reply:
Didn't see Lidel's comment before adding this.

He's correct about the overall purpose of the bar.

However, I think `upgrade` is not correct - it implies a value to the change.

"enable" or something like that would be better.

Reply:
Would also like to avoid terms like "native" and "handler" as we can't necessarily assume users will understand these terms.

Per additional Slack convo w/ @mkarolin , what about this:

`If content is available on the IPFS network (e.g. ipfs:// or ipns://), would you like to load it using Brave's built-in IPFS support?`

Reply:
> If content is available on the IPFS network (e.g. ipfs:// or ipns://), would you like to load it using Brave's built-in IPFS support?

This infobar is shown on a page which is available on the IPFS network, so we should somehow point that we can also redirect for this page once.

Reply:
> @cypt4 — so you mean some variant like:
> 
> `This content is available on the IPFS network (e.g. ipfs:// or ipns://). Would you like to load it using Brave's built-in IPFS support?`
> 
> Also, not sure if enabling here will enable for future pages as well? If yes, could edit to:
> 
> `This content is available on the IPFS network (e.g. ipfs:// or ipns://). Would you like to load it (and future content) using Brave's built-in IPFS support?`

There are 2 enable options - enable only for current page and enable for all pages.

Reply:
@mkarolin — to reduce repetition in the text, might edit to:

`This content is available on the IPFS network. Would you like to load such content using Brave's built-in IPFS support (via ipfs:// or ipns://)? [Always] [Only this time] [No thanks]`

Also: Note the sentence casing on the CTA options.

Reply:
Apologies for spending so much time on this text. It's a major conversion point, so it's important to us to get it right 😄

"such" works for en-uk but not for en-us.

How's this attempt to simplify + clarify:

> This page is available on the IPFS network. Load IPFS content using Brave's built-in support via ipfs:// and ipns:// ?     [Always][Only This Time][No Thanks]

Changes

- s/content/page/, as it's more descriptive, aligns w/ web terminology, and "content" is producer/publisher language not end-user language

- remove parens to be more explicit that the presence of those schemes are what it looks like when the built-in support is being used

Reply:
@autonome — fair point. I'd like to keep the "Would you…" at the start of the sentence to signify this is a question. And the ? after the "ipns://" reads a bit confusing (it appears like it could be part of the scheme). So what if we compromise at:

`This page is available on the IPFS network. Would you like to load pages like this using Brave's built-in IPFS support (via ipfs:// or ipns://)? [Always] [Only This Time] [No Thanks]`

Reply:
If not the above, here's an alt version that's a bit closer to what you suggested:

`This page is available on the IPFS network. Would you like to load IPFS pages using Brave's built-in support (via ipfs:// or ipns://)? [Always] [Only This Time] [No Thanks]`

Reply:
@rmcfadden3 thanks, either of those two wfm!